### PR TITLE
remove termtree dependency

### LINKED ISSUE #11389: Remove the uses of `termtree` dependency
### Is your feature request related to a problem or challenge?

Whatever we can do to make it faster/easier to test and compile datafusion is good, and this includes keeping the dependency chain small. Reducing the dependency chain is also a good way to reduce potential security attacks. 

I noticed in https://github.com/apache/datafusion/pull/11383 that we were upgrading a package called termtree that is only used in a single test. 

### Describe the solution you'd like

Remove the uses of `termtree` package 

https://github.com/search?q=repo%3Aapache%2Fdatafusion%20termtree&type=code

BTW you can run the test in question via:

```shell
cargo test --package datafusion-physical-plan --lib
```

### Describe alternatives you've considered

I think it could be replaced with a `Display` impl and a function that printed out the leaves at the appropriate level

I don't think this requires deep database knowledge, just some rust skills 

### Additional context

cc @avantgardnerio as I think you are the original author

### PR DESCRIPTION
**Pull request by Kev1n8**
## Which issue does this PR close?

Closes #11389

## What changes are included in this PR?

Remove uses of termtree::tree in heap.rs. Directly generate the string instead.

### CODE REVIEW

File: datafusion/physical-plan/src/aggregates/topk/heap.rs [hunk 1]
```diff
@@ -323,29 +323,53 @@ impl<VAL: ValueType> TopKHeap<VAL> {
...
+    fn _tree_print(
+        &self,
+        idx: usize,
+        prefix: String,
+        is_tail: bool,
+        output: &mut String,
+    ) {
+        if let Some(Some(hi)) = self.heap.get(idx) {
+            let connector = if idx != 0 {
+                if is_tail {
```
Comment (alamb) [inline datafusion/physical-plan/src/aggregates/topk/heap.rs:335]: Nice!

File: datafusion/physical-plan/src/aggregates/topk/heap.rs [hunk 2]
```diff
@@ -361,9 +385,9 @@ impl<VAL: ValueType> HeapItem<VAL> {
 impl<VAL: ValueType> Debug for HeapItem<VAL> {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
         f.write_str("bucket=")?;
-        self.map_idx.fmt(f)?;
+        Debug::fmt(&self.map_idx, f)?;
```
Comment (alamb) [inline datafusion/physical-plan/src/aggregates/topk/heap.rs:388]: is this change needed?

File: datafusion/physical-plan/src/aggregates/topk/heap.rs [hunk 3]
```diff
@@ -462,7 +486,7 @@ mod tests {
         let mut heap = TopKHeap::new(10, false);
         heap.append_or_replace(1, 1, &mut map);
 
-        let actual = heap.tree_print();
+        let actual = format!("{heap}");
```
Comment (alamb) [inline datafusion/physical-plan/src/aggregates/topk/heap.rs:489]: Another way to write this is `heap.to_string()`, which might be more "idomatic"

```suggestion
        let actual = heap.to_string()
```

**Review by alamb — COMMENTED** [review]
Thank you @Kev1n8  -- this is a very nice contribution 🙏 

I started the CI -- and once that passes I think this PR could be merged. 

I had a few questions / suggestions but nothing major.

Thanks again!

File: datafusion/physical-plan/src/aggregates/topk/heap.rs (hunk 2 above)
Comment (Kev1n8) [reply datafusion/physical-plan/src/aggregates/topk/heap.rs:388]: I think so. If it's not changed, the compiler will fail and report `fmt` being ambiguous

File: datafusion/physical-plan/src/aggregates/topk/heap.rs (hunk 3 above)
Comment (Kev1n8) [reply datafusion/physical-plan/src/aggregates/topk/heap.rs:489]: sure, I'll fix that

File: datafusion/physical-plan/src/aggregates/topk/heap.rs (hunk 2 above)
Comment (alamb) [reply datafusion/physical-plan/src/aggregates/topk/heap.rs:388]: Makes sense -- thank you

**Review by alamb — APPROVED** [review]
Thanks again @Kev1n8 

FYI @avantgardnerio

**Comment by alamb** [comment]
Thanks again@
